shotover/src/transforms/cassandra/sink_cluster/topology.rs

```diff
@@ -65,133 +65,127 @@ async fn topology_task_process(
     connection_info: &mut TaskConnectionInfo,
     data_center: &str,
 ) -> Result<()> {
-    let (pushed_messages_tx, mut pushed_messages_rx) = unbounded_channel();
+    let force_run_chain = Arc::new(Notify::new());
     connection_info
         .connection_factory
-        .set_pushed_messages_tx(pushed_messages_tx);
+        .set_force_run_chain(force_run_chain);
 
-    let connection = connection_info
+    let version = connection_info.connection_factory.get_version()?;
+    let mut connection = connection_info
         .connection_factory
-        .new_connection(connection_info.address)
+        .new_sink_connection(connection_info.address)
         .await?;
 
-    let version = connection_info.connection_factory.get_version()?;
-
-    let mut nodes = fetch_current_nodes(&connection, connection_info, data_center, version).await?;
+    let mut nodes =
+        fetch_current_nodes(&mut connection, connection_info, data_center, version).await?;
     if let Err(watch::error::SendError(_)) = nodes_tx.send(nodes.clone()) {
         return Ok(());
     }
 
-    let mut keyspaces = system_keyspaces::query(&connection, data_center, version).await?;
-    if let Err(watch::error::SendError(_)) = keyspaces_tx.send(keyspaces.clone()) {
+    let keyspaces = system_keyspaces::query(&mut connection, data_center, version)
+        .await?
+        .0;
+    if let Err(watch::error::SendError(_)) = keyspaces_tx.send(keyspaces) {
         return Ok(());
     }
 
-    register_for_topology_and_status_events(&connection, version).await?;
+    register_for_topology_and_status_events(&mut connection, version).await?;
 
     tracing::info!(
         "Topology task control connection finalized against node at: {:?}",
         connection_info.address
     );
 
+    let mut events = vec![];
     loop {
         // Wait for events to come in from the cassandra node.
         // If all the nodes receivers are closed then immediately stop listening and shutdown the task
-        let pushed_messages = tokio::select! {
-            pushed_messages = pushed_messages_rx.recv() => pushed_messages,
+        tokio::select! {
```

Comment:
We should skip this if events.is_empty otherwise we will await a new message when we already have one  ready to go